dispersy.py

```
@@ -2909,14 +2766,14 @@ def _forward(self, messages):
         elif isinstance(meta.destination, MemberDestination):
             # MemberDestination.candidates may be empty
             result = all(self._send([candidate
-                                            for candidate
-                                            in self._candidates.itervalues()
-                                            if any(candidate.is_associated(message.community, member)
-                                                   for member
-                                                   in message.destination.members)],
-                                           [message])
-                       for message
-                       in messages)
+                                     for candidate
+                                     in message.community.candidates.itervalues()
```

False is already returned when no candidates can be found.  Although this may not immediately be obvious from the code.

The bartercast community doesn't check this though, instead, the request will timeout after a few seconds.  So its not an entirely silent fail.  Still, it can definitely use improvement.  See #84.

[edit: yea, in that case my answer doesn't apply anymore either]